dot/rpc/modules/state_test.go

```diff
@@ -797,89 +799,306 @@ func TestStateModuleGetStorageSize(t *testing.T) {
 }
 
 func TestStateModuleQueryStorage(t *testing.T) {
-	qkvc1 := core.QueryKeyValueChanges{}
-	qkvc1["p1"] = "jimmy"
-	qkvc2 := core.QueryKeyValueChanges{}
-	qkvc2["p2"] = "jimbo"
-
-	hash1 := common.MustHexToHash("0x3aa96b0149b6ca3688878bdbd19464448624136398e3ce45b9e755d3ab61355a")
-	hash2 := common.MustHexToHash("0x3aa96b0149b6ca3688878bdbd19464448624136398e3ce45b9e755d3ab61355a")
-
-	m := map[common.Hash]core.QueryKeyValueChanges{}
-	m[hash1] = qkvc1
-	m[hash2] = qkvc2
-
-	mockCoreAPI := new(mocks.CoreAPI)
-	mockCoreAPI.On("QueryStorage", hash1, hash2, "jimbo").Return(m, nil)
-
-	mockCoreAPIErr := new(mocks.CoreAPI)
-	mockCoreAPIErr.On("QueryStorage", hash1, hash2, "jimbo").Return(nil, errors.New("QueryStorage Error"))
+	t.Parallel()
+	k1 := "0x010204"
+	k2 := "0x090909"
+	v1 := "0x010101"
+	v2 := "0x020202"
+	v3 := "0x030303"
+	v9 := "0x09090909"
 
 	type fields struct {
-		networkAPI NetworkAPI
-		storageAPI StorageAPI
-		coreAPI    CoreAPI
+		storageAPIBuilder func(ctrl *gomock.Controller) StorageAPI
+		blockAPIBuilder   func(ctrl *gomock.Controller) BlockAPI
 	}
 	type args struct {
 		in0 *http.Request
 		req *StateStorageQueryRangeRequest
 	}
-	tests := []struct {
-		name   string
+	tests := map[string]struct {
 		fields fields
 		args   args
 		expErr error
 		exp    []StorageChangeSetResponse
 	}{
-		{
-			name:   "OK Case",
-			fields: fields{nil, nil, mockCoreAPI},
+		"missing start block error": {
+			fields: fields{func(ctrl *gomock.Controller) StorageAPI {
+				return NewMockStorageAPI(ctrl)
+			},
+				func(ctrl *gomock.Controller) BlockAPI {
+					return NewMockBlockAPI(ctrl)
+				}},
```

Comment:
nit use named fields (more readable and the identation looks strange here)

```suggestion
			fields: fields{
				storageAPIBuilder: func(ctrl *gomock.Controller) StorageAPI {
					return NewMockStorageAPI(ctrl)
				},
				blockAPIBuilder: func(ctrl *gomock.Controller) BlockAPI {
					return NewMockBlockAPI(ctrl)
				},
			},
```